### DISCUSSION TITLE
Convergence problem - Solution access

### DISCUSSION THREAD

**Discussion by mkaracelebi** [NREL-SIIP/PowerSimulationsDynamics.jl#266] [Q&A]
I have been using this package in my PhD and thanks for all the effort. I have one small question. I am mostly working on stability problems and it is important to retrieve instable solutions as well. However, sometimes IDA() solver has a convergence problem and terminates the simulation which is normal but is there any way for me to retrieve the solution until the convergence? This might be useful for me to investigate the source of the instability. Thanks a lot for your time.

**Comment by rodrigomha** [comment]
You are correct. We are not storing the solution if the Simulation Failed. You can check the function at line 469 in src/base/simulation.jl, that does not store the solution. If you want you can store the results even if it fail the retcode in DiffEq solve. You would need to modify locally that function and run PSID locally. 

@jd-lara Do you think we should store the solution even if it fails? I don't remember if we are using some logic with isnothing in some parts when the solution does not converge.

**Reply by jd-lara** [reply]
I think this is possible but might require some management of the resulting trace since DiffEq won't do any interpolation if the return error fails. @mkaracelebi please open an issue as a feature request.
